crates/api_common/src/lib.rs

```diff
@@ -485,8 +485,8 @@ pub async fn check_registration_application(
       RegistrationApplication::find_by_local_user_id(conn, local_user_id)
     })
     .await??;
-    if registration.deny_reason.is_some() {
-      return Err(LemmyError::from_message("registration_denied"));
+    if let Some(deny_reason) = registration.deny_reason {
+      return Err(LemmyError::from_message(&deny_reason));
```

Comment:
Most of the deny reasons i see on lemmy.ml are really short, eg "Not an answer". Its going to be very confusing if that is shown as an error message after login. So this should have some message such as: "Registration denied: {{deny_reason}}".

Reply:
I could add that, but it wouldn't be i18n compatible, it'd be always english. This is because the front end, if it can't find that i18n string, just outputs the whole text. So since it can't find the string : "registration_denied: X reason here", it would always be in english.

Reply:
I mean do the localization on the backend, same way we do for email.

Reply:
I'm not fully familiar with how this works. In this [build.rs](https://github.com/LemmyNet/lemmy/blob/main/crates/utils/build.rs), I'd need to add a 2nd src file with all those languages? Seems like there might be a namespace type conflict?

Reply:
K done.